src/scripts/main.js

```diff
@@ -1,3 +1,182 @@
 'use strict';
 
-// write code here
+const tbody = document.querySelector('tbody');
+const thead = document.querySelector('thead');
+
+const pushNotification = (posTop, posRight, title, description, type) => {
+  const body = document.querySelector('body');
+
+  const message = document.createElement('message');
+
+  message.insertAdjacentHTML('afterbegin', `
+  <h2 class = title>${title}</h2>
+  <p>${description}</p>
+  `);
+
+  message.className = `notification ${type}`;
+  message.style.top = `${posTop}px`;
+  message.style.right = `${posRight}px`;
+
+  body.append(message);
+
+  setTimeout(() => {
+    message.remove();
+  }, 2000);
+}; // notifications
+
+tbody.addEventListener('click', (eve) => {
+  for (const tr of [...tbody.children]) {
+    if (tr.className === 'active') {
+      tr.classList.remove('active');
+    }
+  }
+
+  const selected = eve.target.closest('tr');
+
+  selected.classList.add('active');
+}); // selecting rows
+
+thead.addEventListener('click', (eve) => {
+  const sort = (index) => {
+    let sorted = null;
+
+    if (index === 4) {
+      sorted = [...tbody.rows].sort(
+        (a, b) =>
+          a.cells[index].innerHTML.replace(/[^0-9]/g, '')
+          - b.cells[index].innerHTML.replace(/[^0-9]/g, ''));
+    } else {
+      sorted = [...tbody.rows].sort(
+        (a, b) => a.cells[index].innerHTML > b.cells[index].innerHTML ? 1 : -1);
+    }
+
+    tbody.append(...sorted);
+  };
```

Comment:
it's helper function, better to move out the event listener, so it won't be created every time you click on thead